mbed/mbed.py

```diff
@@ -177,6 +177,20 @@ def progress():
     sys.stdout.flush()
     sys.stdout.write('\b')
 
+def show_progress(title, percent, max_width=80):
+    percent = round(float(percent), 2)
+    show_percent = '%.2f' % percent
+    line = str(title) + ' |'
+    bwidth = max_width - len(str(title)) - len(show_percent) - 6
+    for i in range(0, bwidth):
+        line += '#' if i <= (percent / 100 * bwidth) else '-'
+    line += '| ' + show_percent + '%'
+    sys.stdout.write(line+'\r')
+    sys.stdout.flush()
+    sys.stdout.write('\b')
+
+def hide_progress():
+    sys.stdout.write("\033[K\r\b")
```

Comment:
This seems a little magic. Does this work cross platform?

Reply:
I've checked it on Mac OS, Linux, Windows 7 and Windows 10.
Esc[K means Clear line from cursor right

Escape sequences are standardized for years for ANSI/VT-100 compatible terminals. http://ascii-table.com/ansi-escape-sequences-vt-100.php.

Not sure how this would look in a Jenkins log though as it's not a terminal. Any ideas how we could test with Jenkins?

Reply:
Just noticed that on Windows there's a parasite character. Pending minor patch to solve it.

Reply:
@theotherjimmy Magic removed in favour of more standard cursor handling :)